changedetectionio/html_tools.py

```diff
@@ -363,22 +363,41 @@ def extract_json_as_string(content, json_filter, ensure_is_ldjson_info_type=None
 # wordlist - list of regex's (str) or words (str)
 # Preserves all linefeeds and other whitespacing, its not the job of this to remove that
 def strip_ignore_text(content, wordlist, mode="content"):
-    i = 0
-    output = []
     ignore_text = []
     ignore_regex = []
-    ignored_line_numbers = []
+    ignore_regex_multiline = []
+    ignored_lines = []
 
     for k in wordlist:
         # Is it a regex?
         res = re.search(PERL_STYLE_REGEX, k, re.IGNORECASE)
         if res:
-            ignore_regex.append(re.compile(perl_style_slash_enclosed_regex_to_options(k)))
+            res = re.compile(perl_style_slash_enclosed_regex_to_options(k))
+            if res.flags & re.DOTALL or res.flags & re.MULTILINE:
+                ignore_regex_multiline.append(res)
+            else:
+                ignore_regex.append(res)
         else:
             ignore_text.append(k.strip())
 
-    for line in content.splitlines(keepends=True):
-        i += 1
+    for r in ignore_regex_multiline:
```

Comment:
Can you explain what the approach is here so i dont have to manually compile it in my brain?

Reply:
It finds all matches over the whole content using finditer, without splitting it into lines.
For each match it calculates the start/end line number by using splitlines on part of the content before the match end, and the number of lines in the matched content.

Instead of accumulating output lines to an array, the line numbers are added to ignored_lines array which is then converted to a set to remove duplicates. This way multiple filtering methods can mark lines as ignored, and the output is constructed as the last step.